lib/modules/manager/sbt/extract.ts

```diff
@@ -356,10 +385,110 @@ export function extractPackageFile(content: string): PackageFile | null {
     deps: [],
     isMultiDeps: false,
     scalaVersion: null,
-    variables: {},
+    packageFile,
+    ...defaultAcc,
   };
 
   // TODO: needs major refactoring?
   const res = lines.reduce(parseSbtLine, acc);
   return res.deps.length ? res : null;
 }
+
+async function prepareLoadPackageFiles(
+  _config: ExtractConfig,
+  packageFiles: string[]
+): Promise<{
+  variables: Record<
+    string,
+    { val: string; sourceFile: string; lineIndex: number }
+  >;
+  registryUrls: string[];
+}> {
+  let variables: ParseOptions['variables'] = {};
+  let registryUrls: string[] = [MAVEN_REPO];
+  const acc: PackageFile & ParseOptions = {
+    registryUrls,
+    deps: [],
+    variables,
+  };
+
+  for (const packageFile of packageFiles) {
```

Comment:
Seems like we can loop over `packageFiles` once, if only `extractFile` would be able to return variables, dependencies and registry URLs together.

Reply:
I was thinking about the case if it found a line that referenced a variable before it able to resolve.
That's why i read all files before starting to extract.

And it also return `scalaVersion` too that being used as a packageName suffix like `_2.13`